activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -3,10 +3,10 @@
 module ActiveRecord
   module ConnectionAdapters
     class ConnectionSpecification #:nodoc:
-      attr_reader :config, :adapter_method
+      attr_reader :config, :adapter_method, :id
 
-      def initialize(config, adapter_method)
-        @config, @adapter_method = config, adapter_method
+      def initialize(id, config, adapter_method)
+        @config, @adapter_method, @id = config, adapter_method, id
```

Comment:
Readability: find the `id` shifting from being the last arg to first here confusing. I would keep it consistent by always putting it first.